src/ray/util/util.h

```diff
@@ -237,3 +238,64 @@ inline void SetThreadName(const std::string &thread_name) {
   pthread_setname_np(pthread_self(), thread_name.substr(0, 15).c_str());
 #endif
 }
+
+inline std::string GetThreadName() {
+#if defined(__linux__)
+  char name[128];
+  auto rc = pthread_getname_np(pthread_self(), name, sizeof(name));
+  if (rc != 0) {
+    return "ERROR";
+  } else {
+    return name;
+  }
+#else
+  return "UNKNOWN";
+#endif
+}
+
+namespace ray {
+template <typename T>
+class ThreadPrivate {
+ public:
+  template <typename... Ts>
+  ThreadPrivate(Ts &&... ts) : t_(std::forward<Ts>(ts)...) {}
```

Comment:
Why not set the thread id in the constructor?

Reply:
We are not monitoring where it's constructed, but where is used. They can be constructed in the main thread and used in io_worker thread.